meta-facebook/yv35-cl/src/platform/plat_pmic.c

```diff
@@ -0,0 +1,223 @@
+	{ 0x00, 0x00, 0x80, 0x00, 0x00, 0x00 }, // VIN_BULK_UV
+	{ 0x00, 0x00, 0x00, 0x10, 0x02, 0x00 }, // VIN_MGMT_TO_VIN_BUCK_SWITCHOVER
+	{ 0x00, 0x00, 0x00, 0x80, 0x02, 0x00 }, // HIGH_TEMP_WARNING
+	{ 0x00, 0x00, 0x00, 0x20, 0x02, 0x00 }, // VOUT_1V8_PG
+	{ 0x00, 0x00, 0x00, 0x20, 0x02, 0x00 }, // HIGH_CURRENT_WARNING
+	{ 0x00, 0x00, 0x00, 0x00, 0x02, 0xF0 }, // CURRENT_LIMIT_WARNING
+	{ 0x00, 0x00, 0x40, 0x00, 0x00, 0x00 }, // CURRENT_TEMP_SHUTDOWN
+};
+
+static bool is_pmic_error_flag[MAX_COUNT_DIMM][MAX_COUNT_PMIC_ERROR_TYPE];
+
+void start_monitor_pmic_error_thread()
+{
+	printf("Start thread to monitor PMIC error\n");
+
+	monitor_pmic_error_tid =
+		k_thread_create(&monitor_pmic_error_thread, monitor_pmic_error_stack,
+				K_THREAD_STACK_SIZEOF(monitor_pmic_error_stack),
+				monitor_pmic_error_handler, NULL, NULL, NULL,
+				CONFIG_MAIN_THREAD_PRIORITY, 0, K_NO_WAIT);
+	k_thread_name_set(&monitor_pmic_error_thread, "monitor_pmic_error_thread");
+}
+
+void monitor_pmic_error_handler()
+{
+	ipmb_error status = 0;
+	int ret = 0, dimm_id = 0;
+	uint8_t seq_source = 0xFF;
+	memory_write_read_req pmic_req;
+	ipmi_msg pmic_msg;
+
+	// initialize PMIC error flag array
+	memset(is_pmic_error_flag, false, sizeof(is_pmic_error_flag));
+
+	while (1) {
+		// If post isn't complete, BIC doesn't monitor PMIC error
+		if (get_post_status() == false) {
+			k_msleep(MONITOR_PMIC_ERROR_TIME_MS);
+			continue;
+		}
+
+		for (dimm_id = 0; dimm_id < MAX_COUNT_DIMM; dimm_id++) {
+			// Read PMIC error
+			memset(&pmic_req, 0, sizeof(memory_write_read_req));
+			ret = get_dimm_info(dimm_id, &pmic_req.smbus_identifier,
+					    &pmic_req.smbus_address);
+			if (ret < 0) {
+				continue;
+			}
+			pmic_req.intel_id = INTEL_ID;
+			pmic_req.addr_size = PMIC_ADDR_SIZE;
+			pmic_req.addr_value = PMIC_POR_ERROR_LOG_ADDR_VAL;
+			pmic_req.data_len = MAX_LEN_GET_PMIC_ERROR_INFO;
+			pmic_msg = construct_ipmi_message(seq_source, NETFN_NM_REQ,
+							  CMD_SMBUS_READ_MEMORY, SELF, ME_IPMB,
+							  PMIC_READ_DATA_LEN, (uint8_t *)&pmic_req);
+
+			// Double check post status before send IPMB to get PMIC error
+			if (get_post_status() == false) {
+				continue;
+			}
+			status = ipmb_read(&pmic_msg, IPMB_inf_index_map[pmic_msg.InF_target]);
+			if (status == IPMB_ERROR_SUCCESS) {
+				// Compare error pattern, if status change add SEL to BMC and update record
+				ret = compare_pmic_error(dimm_id, pmic_msg.data);
+				if (ret < 0) {
+					continue;
+				}
+
+			} else {
+				printf("%s(): Failed to get PMIC error, dimm %s bus %d addr 0x%x status 0x%x\n",
+				       __func__, dimm_lable[dimm_id], pmic_req.smbus_identifier,
+				       pmic_req.smbus_address, status);
+				continue;
+			}
+		}
+
+		k_msleep(MONITOR_PMIC_ERROR_TIME_MS);
+	}
+}
+
+int get_dimm_info(uint8_t dimm_id, uint8_t *bus, uint8_t *addr)
+{
+	int ret = 0;
+
+	if ((bus == NULL) || (addr == NULL)) {
+		return -1;
+	}
+
+	if ((dimm_id < 0) || (dimm_id >= MAX_COUNT_DIMM)) {
```

Comment:
dimm_id is a uint8_t so can never be <0.  Please fix this test to remove the <0 test.  Thanks!